charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py

```diff
@@ -147,6 +148,11 @@ def patch_json_for_multicluster_configuration(content, multicluster_key):
     return content
 
 
+def patch_json_set_timezone_as_variable(content):
+    # content is no more in json format, so we have to replace using regex
+    return re.sub(r'\"timezone\":.*$', '"timezone": "\{\{ .Value.grafana.defaultDashboardsTimezone \}\}",', content, flags=re.IGNORECASE|re.MULTILINE)
```

Comment:
Seems like this could go wrong, can we be more specific with the regex or parse a sub-section of the input file as json?

Reply:
Yes, the regex was a bit weak.
I'd have liked to manipulate directly the json object, but '\{\{' is not an allowed json sequence, as '\' must be escaped all the times => '\\{\\{'. However the '\\{\\{' is not converted correctly by yaml_str_repr on line 97, as it replaces exactly '\{\{'.
I didn't want to change too much yaml_str_repr, nor introduce a complex function as patch_json_for_multicluster_configuration.